python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -878,23 +878,6 @@ def _apply_filter_to_query(
                 isinstance(event_records_filter.asset_key, AssetKey),
                 "Asset key must be set in event records filter to filter by tags.",
             )
-            if self.supports_intersect:
```

Comment:
Am I understanding correctly that the caller of this always filters after the fact...sometimes on a list that's already filtered, sometimes on one that isn't?

Otherwise, how could `self.supports_intersect` and `not self.supports_intersect` return the same results?

Reply:
yeah it's hard to follow. _apply_filter_to_query is only called by _get_event_records, which also checks supports_intersect and if it's not supported, does the filtering via _apply_tags_table_joins